extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts

```diff
@@ -0,0 +1,518 @@
+
+		// Information bubble title to be set depening on type of input
+		let information = this.modelView.modelBuilder.button().withProperties<azdata.ComponentWithIconProperties>({
+			iconPath: IconPathHelper.information,
+			width: '12px',
+			height: '12px',
+			enabled: false
+		}).component();
+
+		if (type === 'enum') {
+			// If type is enum, component should be drop down menu
+			let valueBox = this.modelView.modelBuilder.dropDown().withProperties<azdata.DropDownProperties>({
+				values: [], //TODO,
+				value: '', //TODO
+				CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+			}).component();
+			valueContainer.addItem(valueBox, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+			this.disposables.push(
+				valueBox.onValueChanged(() => {
+					this.engineSettingUpdates!.set(name, String(valueBox.value));
+
+				})
+			);
+
+			information.updateProperty('title', loc.optionsSetting('enums'));	//TODO
+		} else if (type === 'bool') {
+			// If type is bool, component should be checkbox to turn on or off
+			let valueBox = this.modelView.modelBuilder.checkBox().withProperties<azdata.CheckBoxProperties>({
+				label: loc.on,
+				checked: true, //TODO
+				CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+			}).component();
+			valueContainer.addItem(valueBox, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+			this.disposables.push(
+				valueBox.onChanged(() => {
+					if (valueBox.checked) {
+						this.engineSettingUpdates!.set(name, 'on');
+					} else {
+						this.engineSettingUpdates!.set(name, 'off');
+					}
+				})
+			);
+
+			information.updateProperty('title', loc.optionsSetting('on,off'));	//TODO
+		} else if (type === 'string') {
+			// If type is string, component should be text inputbox
+			// How to add validation: .withValidation(component => component.value?.search('[0-9]') == -1)
+			let valueBox = this.modelView.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+				readOnly: false,
+				value: '', //TODO
+				CSSStyles: { 'margin-bottom': '15px', 'min-width': '50px', 'max-width': '200px' }
+			}).component();
+			valueContainer.addItem(valueBox, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+			this.disposables.push(
+				valueBox.onTextChanged(() => {
+					this.engineSettingUpdates!.set(name, valueBox.value!);
+				})
+			);
+
+			information.updateProperty('title', loc.optionsSetting(loc.optionsSetting('[A-Za-z._]+')));	//TODO
+		} else {
+			// If type is real or interger, component should be inputbox set to inputType of number. Max and min values also set.
+			let valueBox = this.modelView.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+				readOnly: false,
+				min: 0, //TODO
+				max: 10000,
+				validationErrorMessage: loc.outOfRange('min', 'max'), //TODO
```

Comment:
localize min/max?

Reply:
These should be set once knowing the min/max values from the engine settings. Right now just filler.